syntax = "proto3";

package proto;

enum Side {
  BUY = 0;
  SELL = 1;
}

enum OrderType {
  LIMIT = 0;
  MARKET = 1;
}

message OrderRequest {
  string cl_ord_id = 1;
  string exch      = 2;
  string symbol    = 3;
  Side side        = 4;
  OrderType type   = 5;
  double qty       = 6;
  double price     = 7; // optional for market
  uint64 timestamp_us = 8;
}

enum OrderEventType {
  ACK = 0;
  FILL = 1;
  CANCEL = 2;
  REJECT = 3;
}

message OrderEvent {
  string cl_ord_id  = 1;
  string exch       = 2;
  string symbol     = 3;
  OrderEventType event_type = 4;
  double fill_qty   = 5;
  double fill_price = 6;
  string text       = 7;
  uint64 timestamp_us = 8;
  string exch_order_id = 9;  // Exchange-assigned order ID
}


